dotcom-rendering/src/lib/alternate-lang-links.ts

```diff
@@ -0,0 +1,65 @@
+import type { Edition } from './edition';
+import {
+	editionList,
+	getEditionFromPageId,
+	isNetworkFront,
+	splitEditionalisedPage,
+} from './edition';
+
+const linkTemplate = (baseUrl: string, pageId: string, lang: string): string =>
+	`<link rel="alternate" href="${baseUrl}/${pageId}" hreflang="${lang}" />`;
+
+const collectAlternateLangLinks = ({
+	baseUrl,
+	path = [],
+	predicate = () => true,
+}: {
+	baseUrl: string;
+	path?: string[];
+	predicate?: (edition: Edition) => boolean;
+}): string[] => {
+	return editionList.reduce<string[]>((acc, edition) => {
+		if (edition.langLocale && predicate(edition)) {
+			return [
+				...acc,
+				linkTemplate(
+					baseUrl,
+					[edition.pageId, ...path].join('/'),
+					edition.langLocale,
+				),
+			];
+		}
+		return acc;
+	}, []);
+};
+
+const createAlternateLangLinks = (
+	baseUrl: string,
+	pageId: string,
+): string[] => {
+	if (isNetworkFront(pageId)) {
+		const pageEdition = getEditionFromPageId(pageId);
+		const hasLangLocale = pageEdition?.langLocale;
```

Comment:
Maybe we want to make sure this is a `boolean`?

```suggestion
		const hasLangLocale = !!pageEdition?.langLocale;
```

Reply:
Thanks for the spots!